File: GestionMorgueFinal/src/app/@core/backend/common/model/CertificatMedicoLegal.ts
import {Medecins} from './Medecins';
import {Decedes} from './Decedes';

export class CertificatMedicoLegal {
id: number;
medecin: Medecins ;
declarant: string;
address: string;
cin: string;
declaration: Date;
defunt: Decedes ;
  constructor() {
  }
}

File: GestionMorgueBack/src/main/java/com/akveo/bundlejava/model/CertificatMedicoLegal.java

package com.akveo.bundlejava.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.OneToOne;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.GenerationType;
import java.io.Serializable;
import java.util.Date;
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class CertificatMedicoLegal implements Serializable {
    @Id
    @GeneratedValue (strategy = GenerationType.IDENTITY)
    private long id;
    @OneToOne
    private Medecins medecin;
    private String declarant;
    private String address;
    private String cin;
    private Date declaration;
    @OneToOne
    private Decedes defunt;
}
